### PR TITLE
Recipe button fix

### LINKED ISSUE #101: Allow for user to navigate to full payed for or uploaded recipes:
- [ ] change button contents but not full disable when a user owns a recipe

### PR DESCRIPTION
**Pull request by nathanielmillard**
## Pull request type
Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Other (please describe):

## Problem/feature
- #101 

## Solution
- closes #101 but allow for dynamic rendering of what is in the button box but not the functionality
- added in updates of the user in local storage on page loads for recipe book and main page

## Checklist
- [x] Code compiles correctly
- [ ] Tests for the changes have been added (for bug fixes/features)
- [x] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: src/components/MainPage/MainPage.js [hunk 1]
```diff
@@ -28,6 +28,18 @@ function MainPage(props) {
...
+    let userInfo = JSON.stringify(response.data.getUser)
+    localStorage.setItem('user', userInfo)
+  }
+
+  useEffect(() => {
+    let user = JSON.parse(localStorage.getItem('user'));
+    if (user) {
+      updateUser(user.id)
+    }
+  })
```
Comment (hashmaster3k) [inline src/components/MainPage/MainPage.js:42]: Correct me if I am wrong, this is checking if a user exists (i.e logged in) and if they are, update the localStorage with the most up to date information from the DB?

File: src/components/RecipeCard/RecipeCard.js [hunk 2]
```diff
@@ -1,52 +1,30 @@
...
+    buttonText = "Give N' Get Recipe"
   }
 
-  let rating;
-  if (props.rating != null) {
+  if (props.rating !== null) {
     rating = props.rating + " out of 5";
   } else {
     rating = "Not yet rated";
   }
```
Comment (hashmaster3k) [inline src/components/RecipeCard/RecipeCard.js:27]: Thanks for fixing this. I apologize for the spaghetti code. I know it was logic heavy and not idea but was trying to get it working.

**Review by hashmaster3k — APPROVED** [review]

File: src/components/MainPage/MainPage.js (hunk 1 above)
Comment (nathanielmillard) [reply src/components/MainPage/MainPage.js:42]: Yep!

File: src/components/RecipeCard/RecipeCard.js (hunk 2 above)
Comment (nathanielmillard) [reply src/components/RecipeCard/RecipeCard.js:27]: No worries at all! I am glad it was up and running and could comb through it later :)

**Pull request merged**